clock/interface.go

```diff
@@ -0,0 +1,11 @@
+package clock
+
+import (
+	"time"
+)
+
+// Clock is an interface that provides a time functions for LND packages.
+// This is useful during testing when a concrete time reference is needed.
+type Clock interface {
```

Comment:
Couldn't a closure of `now func() time.Time` or a lighter interface _within_ the package that this is actually used have sufficed? I'm not sure such a bare interface warrants a new package, as the interace itself doesn't specify any behavior.

Reply:
If there's a need for a "test clock" (stubbed out now) with state driven behavior then if now() is simply a closure, then state needs to be global (apart from trivial cases).
Clock can also be extended with other stubbed out functions like time.After() <- time.Time etc.

Reply:
I like the part where you can also get `time.After` from here. We keep re-inventing this in unit tests.

Reply:
I can add `time.After` now or later as needed.

Reply:
I'd say only add it when we need it. I like the `Clock` interface. In the mpp pr, I also made a clock (that also has `tickAfter`)

Reply:
Ok, now we have it. Nice